### PR TITLE
Transitioning from Bootstrap to Semantic-UI React

### LINKED ISSUE #3321: Transitioning from Bootstrap to Semantic UI React
### Introduction:
Website styling and design is an integral part of front end web development and web developers are expected to make available a flawless web design to optimize user experience and interaction.
Bootstrap has been the leading library in web styling and favored by many developers untill now. Semantic UI React and other modern design libraries are reducing the complexity of web styling by providing advanced elements and features as we'll discuss in this article
### key takeaways:
At the end of this article, the reader is expected to understand and implement the following
i. The fundamentals of front end web design libraries
ii. Limitations of bootstrap
iii. Efficiencies and advanced features of Semantic UI React
iv. Implementing a simple web page with semantic UI React elements
v. Tricks and hacks in Semantic UI React
### Reference
https://www.tabnine.com/code/javascript/modules/semantic-ui-react

### PR DESCRIPTION
**Pull request by NyongToro**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

#### Topic suggestion this closes

This closes #3321.

### CODE REVIEW

**Comment by WanjaMIKE** [comment]
Plagiarism check done.

**Comment by NyongToro** [comment]
@Dawe-7 
It's been 12days any update yet?

**Comment by Dawe-7** [comment]
> @Dawe-7
> It's been 12days any update yet?

@NyongToro working on it.

File: content/articles/transitioning-from-bootstrap-to-semantic-ui-react/index.md [hunk 1]
```diff
@@ -0,0 +1,622 @@
...
+
+#### Getting started with Semantic-UI React
+
+Semantic-UI is a modern web design and styling framework. It provides efficient and customizable elements such as
+
+1. Avatars
+2. Modals
+3. Popups
+4. Cards and Icons, etc.
+   Using semantic-UI in your `React` application is simple. But for absolute beginners we will go through the steps to get you familiar with the usage.
```
Comment (Dawe-7) [inline content/articles/transitioning-from-bootstrap-to-semantic-ui-react/index.md]:
```suggestion
Using semantic-UI in your `React` application is simple. But for absolute beginners,  we will go through the steps to get you familiar with the usage.
```

File: content/articles/transitioning-from-bootstrap-to-semantic-ui-react/index.md [hunk 2]
```diff
@@ -0,0 +1,622 @@
...
+We now have access to the library and we will use it to create a sample page. The page will contain the following elements
+
+- Navbar
+- Avatar
+- Header Texts
+- Popup
+- Button and Input Fields
+- Card
+
+To achieve this, we will create two files `Navbar.js` and `Body.js` thereafter we will import the above elements and add it to our web page. be sure to create the files in your `src` directory
```
Comment (Dawe-7) [inline content/articles/transitioning-from-bootstrap-to-semantic-ui-react/index.md:377]:
```suggestion
To achieve this, we will create two files `Navbar.js` and `Body.js`.  Afterwards,  we will import the above elements and add them  to our web page. Be sure to create the files in your `src` directory.
```

File: content/articles/transitioning-from-bootstrap-to-semantic-ui-react/index.md [hunk 3]
```diff
@@ -0,0 +1,622 @@
...
+        <Body />
+      </div>
+    );
+}
+export default App;
+```
+
+#### Customizing Elements in Semantic-UI React
+
+Semantic-UI components and elements are designed with default values in color, size, orientation and layout. with the aid a `CSS` file you can customize each Semantic-UI element to your desired specifications. Below is an example of element customization using CSS.
```
Comment (Dawe-7) [inline content/articles/transitioning-from-bootstrap-to-semantic-ui-react/index.md]:
```suggestion
Semantic-UI components and elements are designed with default values in color, size, orientation, and layout. With the aid of a `CSS`  file, you can customize each Semantic-UI element to your desired specifications. Below is an example of element customization using CSS.
```

File: content/articles/transitioning-from-bootstrap-to-semantic-ui-react/index.md [hunk 4]
```diff
@@ -0,0 +1,622 @@
...
+
+1. Semantic-UI design Library is light-weight and easy to use: meaning it does not increase the package size of your project.
+2. Semantic-UI contains hundreds of customizable icons: all you have to do is import whichever icon into your project.
+3. Semantic-UI has high efficiency and compilation rate.
+4. Semantic-UI is easy to use and understand.
+5. Semantic-UI is used by numerous tech companies, such as Snapchat, Google Cloud Partners etc.
+
+### Conclusion
+
+We have compared both Bootstrap and Semantic-UI design library in this article. The verdict is out and is up to you to decide which one best suit your web needs, but I strongly recommend Semantic-UI. I hope you find this article helpful in your web development journey.
```
Comment (Dawe-7) [inline content/articles/transitioning-from-bootstrap-to-semantic-ui-react/index.md:616]:
```suggestion
We have compared both Bootstrap and Semantic-UI design libraries in this article. The verdict is out and is up to you to decide which one best suits your web needs, but I strongly recommend Semantic-UI. I hope you find this article helpful in your web development journey.
```

File: content/articles/transitioning-from-bootstrap-to-semantic-ui-react/index.md [hunk 5]
```diff
@@ -0,0 +1,622 @@
...
+The Semantic-UI package comes with numerous modern features. A few of which includes the following
+
+1. Simplified Error handling and debugging
+2. High level theming variables
+3. Advanced elements such as Accordions, Dividers, Segments, Menu, etc.
+4. Easy to use APIs
+5. Customizable layouts and orientation.
+
+#### Why you should Choose Semantic-UI React
+
```
Comment (Dawe-7) [inline content/articles/transitioning-from-bootstrap-to-semantic-ui-react/index.md:605]:
```suggestion
```

**Comment by NyongToro** [comment]
@Dawe-7 I apologize for the empty spaces. It was caused by an extension installed in my code editor. I have resolved all the issues you raised.

**Comment by hectorkambow** [comment]
@Dawe-7 just wanted to get a status on this PR 👍

**Review by Dawe-7 — APPROVED** [review]
@NyongToro , great article. Kindly go through the changes I've made to improve on your future articles.

@hectorkambow, this article is ready for final review.

**Comment by hectorkambow** [comment]
Overall great work Gents @NyongToro 

cc @Dawe-7 

The tutorial was thorough and easy to follow along 🚀 

Made some minor edits on my end as well for flow and format. 


Please be sure to review any edits made by me or your peer reviewer to improve future articles and reviews. 

cc @Dawe-7 please do the same 👍

**Pull request merged**